apps/tools/contentful-role-permissions/pages/api/update-role-permissions.ts

```diff
@@ -0,0 +1,110 @@
+import slugify from '@sindresorhus/slugify'
+import {
+  ContentTypeProps,
+  Role,
+  RoleProps,
+  TagProps,
+} from 'contentful-management'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import {
+  applyAssetPolicies,
+  applyEditEntryPolicies,
+  applyReadOnlyEntryPolicies,
+  extractInitialCheckboxStateFromRolesAndContentTypes,
+  extractInititalReadonlyCheckboxStateFromRolesAndContentTypes,
+  getAllContentTypesInAscendingOrder,
+  getAllRoles,
+  getContentfulManagementApiClient,
+} from '../../utils'
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const data = JSON.parse(req.body) as {
+    checkboxState: ReturnType<
+      typeof extractInitialCheckboxStateFromRolesAndContentTypes
+    >
+    readonlyCheckboxState: ReturnType<
+      typeof extractInititalReadonlyCheckboxStateFromRolesAndContentTypes
+    >
+    roleNamesThatCanReadAllAssets: string[]
+    tags: TagProps[]
+  }
```

Comment:
Would consider refactoring this into explicit types and interfaces.

Reply:
Done :)